ngraph_bridge/ngraph_encapsulate_clusters.cc

```diff
@@ -75,30 +75,287 @@ static void AddInput(NodeDef* dst, StringPiece src_name, int src_slot) {
 
 Status EncapsulateClusters(
     Graph* graph, int graph_id, FunctionDefLibrary* fdeflib,
-    std::unordered_map<std::string, std::string> device_config,
-    AOTInfo aot_info) {
-  // A map from cluster indices to the expected device name for nodes
-  // in that cluster.
-  std::map<int, std::string> device_name_map;
-
-  // We *should* eventually have a way of monitoring the device and the backend
-  // together
-  std::map<int, std::string> backend_name_map;
-
-  // As we build the graph we will be tracking the.. TODO(amprocte): finish
-  // this comment.
-  std::map<std::tuple<int, int>, std::tuple<int, int>> output_remap_map;
-  std::map<std::tuple<int, int, int>, int> input_remap_map;
-  std::map<std::tuple<int, std::string, int>, string> input_rename_map;
-
-  // A map from cluster indices to a vector of input data types.
-  std::map<int, std::vector<std::tuple<int, int, DataType>>> cluster_input_map;
-  // A map from cluster indices to a vector of output data types.
-  std::map<int, std::vector<DataType>> cluster_output_dt_map;
-
-  // A map from cluster indices to corresponding NGraphEncapsulate nodes.
-  std::map<int, Node*> cluster_node_map;
+    const std::unordered_map<std::string, std::string>& device_config,
+    const AOTInfo& aot_info) {
+  Encapsulator enc(graph);
+  NGRAPH_VLOG(3) << "Running AnalysisPass in EncapsulateClusters";
+  TF_RETURN_IF_ERROR(enc.AnalysisPass());
+  NGRAPH_VLOG(3) << "Running RewritePass in EncapsulateClusters";
+  TF_RETURN_IF_ERROR(enc.RewritePass(fdeflib, graph_id, device_config));
+  NGRAPH_VLOG(3) << "Performing AOT in EncapsulateClusters";
+  TF_RETURN_IF_ERROR(PerformAOTOnEncapsulates(graph, aot_info));
+
+  set<int> newly_created_cluster_ids;
+  TF_RETURN_IF_ERROR(enc.NewClusterIds(newly_created_cluster_ids));
```

Comment:
Will CreateClusterIDs be a better name?

Reply:
It does not created new ids. it just returns the new ones that Analysis pass created.
Its more of a "GetNewlyCreatedClusterIDs"

Reply:
Come to think of it, thats what I named my argument (newly_created_cluster_ids) 🤔

Reply:
So may be GetNewClusterIDs(newly_created_cluster_ids)

Reply:
Changed to new name.